draft-specifications/Netlayer-Unix-Domain-Socket.md

```diff
@@ -0,0 +1,123 @@
+  
+- `server-challenge`: This is like the `client-challenge` a ByteArray containing
+  some strong randomly generated noise which the netlayer will sign to prove its
+  identity.
+  
+- `client-challenge-response`: This is `uds:signature` struct. Please see its
+  definition below.
+
+When the netlayer receives the `uds:server-response` message, it MUST check
+the signature of the client-challenge that it sent against the one provided in
+`client-challenge-response`. The signature MUST be valid for the provided
+server `pubkey`. If the signature is not valid for the `pubkey`, the connection
+MUST be closed. Provided the signature is valid, the netlayer MUST then
+response with a `uds:signature` signing the `server-challenge` provided with its
+designator key.
+
+### uds:signature
+
+```
+<uds:signature payload     ; ByteArray of data being signed
+               signature>  ; List representing signature in the same format as CapTP
+```
+
+This is used both within `uds:server-response` and on it's own by the netlayer
+when signing the `server-challenge`. It includes the following values:
+
+- `payload`: The data being signed.
+- `signature`: A cryptographic signature formatted as a List according to the
+  way CapTP formats signatures. i.e.
+  
+  https://github.com/ocapn/ocapn/blob/main/draft-specifications/CapTP%20Specification.md#signature
+
+When the server receives this in response to its `uds:server-response` message,
+it MUST check that the signature is valid for the key encoded in the OCapN
+peer locator designator. If the signature is valid for the key, the server
+can then finish any setup and provide introductions upon request both to and from
+this netlayer. If the signature is invalid the connection MUST be closed and
+no introductions to the netlayer MUST occur.
+
+### uds:new-connection
+
+```
+<uds:new-connection from  ; OCapN Locator of sender's peer Locator 
+                    to>   ; OCapN Locator of receipient's peer Locator
+```
+
+This is used by the netlayer to ask the introduction to perform an introduction
+to another peer. The operation consists of:
+
+- `from`: OCapN Locator of the peer opening the connection.
+- `to`: OCapN Locator of the peer the connection is being opened to.
+
+Once this message has been written to the socket to the introduction server, the
+next message MUST be `SCM_RIGHTS` message with a file descriptor to a Unix domain
+socket that the sender has created. The introduction server will then send this
+exact same `uds:new-connection` message to the peer address in `to`, and
+followed by the SCM_RIGHTS message with the socket it read.
+
+SCM_RIGHTS messages have several parts, one of which is a body, which is not
+used or important to the netlayer, so the data contained within it is ignored.
+The file descriptor itself is placed within the ancillary data in the socket
+control message, these and sent with `sendmsg` and recieved with `recvmsg` For
```

Comment:
Period after "control message" rather than a comma. Then start the next sentence with "The control message is sent using the `sendmsg` libc function and received with `recvmsg`." (note the typo fix for "received")